conductor-client/src/main/kotlin/com/openlattice/collaborations/PostgresCollaborationDatabaseService.kt

```diff
@@ -0,0 +1,250 @@
+package com.openlattice.collaborations
+
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.query.Predicate
+import com.hazelcast.query.Predicates
+import com.openlattice.assembler.AssemblerConfiguration
+import com.openlattice.authorization.AclKey
+import com.openlattice.authorization.AuthorizationManager
+import com.openlattice.authorization.DbCredentialService
+import com.openlattice.authorization.Permission
+import com.openlattice.collaborations.mapstores.ProjectedTablesMapstore
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.organizations.OrganizationDatabase
+import com.openlattice.organizations.mapstores.TABLE_ID_INDEX
+import com.openlattice.organizations.roles.SecurePrincipalsManager
+import com.openlattice.postgres.PostgresProjectionService
+import com.openlattice.postgres.external.DatabaseQueryManager
+import com.openlattice.postgres.external.ExternalDatabaseConnectionManager
+import com.openlattice.postgres.external.ExternalDatabasePermissioningService
+import com.openlattice.postgres.external.Schemas
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class PostgresCollaborationDatabaseService(
+        hazelcast: HazelcastInstance,
+        private val dbQueryManager: DatabaseQueryManager,
+        private val externalDbConnMan: ExternalDatabaseConnectionManager,
+        private val authorizations: AuthorizationManager,
+        private val externalDbPermissioner: ExternalDatabasePermissioningService,
+        private val spm: SecurePrincipalsManager,
+        private val dbCreds: DbCredentialService,
+        private val assemblerConfiguration: AssemblerConfiguration
+) : CollaborationDatabaseManager {
+
+    private val collaborations = HazelcastMap.COLLABORATIONS.getMap(hazelcast)
+    private val organizationDatabases = HazelcastMap.ORGANIZATION_DATABASES.getMap(hazelcast)
+    private val externalTables = HazelcastMap.ORGANIZATION_EXTERNAL_DATABASE_TABLE.getMap(hazelcast)
+    private val externalColumns = HazelcastMap.ORGANIZATION_EXTERNAL_DATABASE_COLUMN.getMap(hazelcast)
+    private val projectedTables = HazelcastMap.PROJECTED_TABLES.getMap(hazelcast)
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(PostgresCollaborationDatabaseService::class.java)
+    }
+
+    override fun getDatabaseInfo(collaborationId: UUID): OrganizationDatabase {
+        return organizationDatabases[collaborationId]!!
+    }
+
+    override fun createCollaborationDatabase(collaborationId: UUID) {
+        val dbName = ExternalDatabaseConnectionManager.buildDefaultCollaborationDatabaseName(collaborationId)
+        val oid = dbQueryManager.createAndInitializeCollaborationDatabase(collaborationId, dbName)
+
+        organizationDatabases[collaborationId] = OrganizationDatabase(oid, dbName)
+    }
+
+    override fun deleteCollaborationDatabase(collaborationId: UUID) {
+        dbQueryManager.dropDatabase(getDatabaseInfo(collaborationId).name)
```

Comment:
On line 53, `organizationDatabases` is updated with the new collaboration database.

```
organizationDatabases[collaborationId] = OrganizationDatabase(oid, dbName)
```

Don't you have to remove from `organizationDatabases` here?

Reply:
ooh good call, yeah